app/models/concerns/foreman_build_history/host_extensions.rb

```diff
@@ -7,6 +7,7 @@ module HostExtensions
       after_build :set_build_start
       before_provision :set_build_finish
       before_save :build_history_handle_host_creation
+      has_one :build_history_facet, :foreign_key => 'host_id'
```

Comment:
Yeah, that probably was it. We should make facts more reliable to reloading, then.